front/src/main.js

```diff
@@ -15,14 +16,20 @@ new Vue({
   router,
   store,
   vuetify,
+  i18n,
   created () {
     // load data from localStorage to vuex (on f5)
-    const userString = localStorage.getItem('user');
+    const userString = localStorage.getItem('user'),
+          language = localStorage.getItem('language');
     if (userString) {
       const userData = JSON.parse(userString);
       this.$store.commit('SET_USER_DATA', userData);
     }
 
+    if (language) {
+      this.$store.dispatch('locale/setLanguage', language);
+    }
+
```

Comment:
store - is not only about get/set some scope of variables. it is about data at all. and all manipulation with data state should be placed in store. If no, what the difference between store and file with global variables? for these manipulations with data we have instruments like actions and mutations. As I said before, it is more flux way to relocate logic of initialization of language into store in action like INIT_LANGUAGE, or in other way, but all logic of storing data is not in components, it is right place - in store. components - are only about user interection

Reply:
I have added INIT_LANGUAGE in the mutation and action initLanguage.  Which method is preferable to use $ store.dispatch or $ store.commit ?

Reply:
you can use mutations from components, usually actions we add for asynchronous operations because mutations are synchronous only

Reply:
Thanks